src/Moryx.Orders.Web/src/app/dialogs/create-dialog/create-dialog.component.ts

```diff
@@ -237,4 +249,33 @@ export class CreateDialogComponent implements OnInit {
     }
     if (!failed) this.dialog.close();
   }
+
+  primaryActionLabel = computed(() => {
+    return this.primaryAction() === 'ADD_CREATE'
+      ? this.translate.instant(TranslationConstants.CREATE_DIALOG.CREATE)
+      : this.translate.instant(TranslationConstants.CREATE_DIALOG.ADD);
+  });
+
+  canAdd    = () => !this.isLoading() && this.canAddOperation();
+
+  canCreate = () => !this.isLoading() && (this.operations().length > 0 || this.canAddOperation());
+
+  canRun = (a: Action) => (a === 'ADD_CREATE' ? this.canCreate() : this.canAdd());
+
+  dropdownDisabled = () => !this.canRun('ADD_CREATE') && !this.canRun('ADD_ONLY');
+
+  onPrimaryClick = async () => { await this.performAction(this.primaryAction()); };
+
+  onSelectAction = (a: Action) => { this.primaryAction.set(a); };
+
+  isSelected     = (a: Action) => this.primaryAction() === a;
+
+  private async performAction(a: Action): Promise<void> {
+    if (a === 'ADD_CREATE') {
+      if (this.canAddOperation()) this.addOperation();
+      if (this.operations().length > 0) await this.create();
+      return;
+    }
+    if (this.canAddOperation()) this.addOperation();
```

Comment:
either make this if-else and remove the return or use the early return and remove the second if, this is half of both options 😅